### DISCUSSION TITLE
Discussion: Inheritance by container than hierarchy

### DISCUSSION THREAD

**Discussion by vbcodec** [dotnet/csharplang#948] [General]
```
    Public Class BaseLike
        Public Exposable Property prop1 As String
    End Class

    Public Class DerivedLike
        Dim Exposed X As New BaseLike
    End Class

    Dim o = New DerivedLike
    o.prop1 = "It works"

```

Could be very helpful to avoid creating 'bridges' while using wrappers.

**Comment by jnm2** [comment]
Looking for https://github.com/dotnet/vblang?

**Comment by vbcodec** [comment]
https://github.com/dotnet/roslyn  :
 https://github.com/dotnet/csharplang for features that affect both languages

**Comment by yaakov-h** [comment]
I see you've provided VB syntax to a C# proposal... but shouldn't this start off in dotnet/coreclr?

Or are you after the compiler rewriting such classes into magic goop?

**Comment by iam3yal** [comment]
This is basically traits or proxy types so related to #862.
